src/code/z_actor.c

```diff
@@ -221,180 +221,191 @@ void Actor_ProjectPos(PlayState* play, Vec3f* src, Vec3f* xyzDest, f32* cappedIn
+
+        reticle->color.r = reticleColor->inner.r;
+        reticle->color.g = reticleColor->inner.g;
+        reticle->color.b = reticleColor->inner.b;
+    }
+}
+
+void Target_SetNaviState(TargetContext* targetCtx, Actor* actor, s32 actorCategory, PlayState* play) {
+    TargetColor* naviColor = &sTargetColorList[actorCategory];
+
+    targetCtx->naviPos.x = actor->focus.pos.x;
+    targetCtx->naviPos.y = actor->focus.pos.y + (actor->targetArrowOffset * actor->scale.y);
+    targetCtx->naviPos.z = actor->focus.pos.z;
+
+    targetCtx->naviInnerColor.r = naviColor->inner.r;
+    targetCtx->naviInnerColor.g = naviColor->inner.g;
+    targetCtx->naviInnerColor.b = naviColor->inner.b;
+    targetCtx->naviInnerColor.a = naviColor->inner.a;
+    targetCtx->naviOuterColor.r = naviColor->outer.r;
+    targetCtx->naviOuterColor.g = naviColor->outer.g;
+    targetCtx->naviOuterColor.b = naviColor->outer.b;
+    targetCtx->naviOuterColor.a = naviColor->outer.a;
+}
+
+void Target_Init(TargetContext* targetCtx, Actor* actor, PlayState* play) {
+    targetCtx->naviHoverActor = NULL;
+    targetCtx->lockOnActor = NULL;
+    targetCtx->naviMoveProgressFactor = 0.0f;
+    targetCtx->forcedTargetActor = NULL;
     targetCtx->bgmEnemy = NULL;
-    targetCtx->unk_4B = 0;
-    targetCtx->unk_4C = 0;
-    Actor_SetNaviToActor(targetCtx, actor, actor->category, play);
-    func_8002BE98(targetCtx, actor->category, play);
+    targetCtx->rotZTick = 0;
+    targetCtx->lockOnIndex = 0;
+    Target_SetNaviState(targetCtx, actor, actor->category, play);
+    Target_InitReticle(targetCtx, actor->category, play);
 }
 
-void func_8002C124(TargetContext* targetCtx, PlayState* play) {
-    Actor* actor = targetCtx->targetedActor;
+void Target_Draw(TargetContext* targetCtx, PlayState* play) {
+    Actor* actor = targetCtx->lockOnActor;
 
     OPEN_DISPS(play->state.gfxCtx, "../z_actor.c", 2029);
 
-    if (targetCtx->unk_48 != 0) {
-        TargetContextEntry* entry;
+    if (targetCtx->reticleFadeAlphaControl != 0) {
+        LockOnReticle* reticle;
         Player* player;
-        s16 spCE;
+        s16 alpha;
         f32 temp1;
-        Vec3f projTargetCenter;
-        s32 spB8;
-        f32 projTargetCappedInvW;
-        s32 spB0;
-        s32 spAC;
-        f32 var1;
-        f32 var2;
+        Vec3f projectedPos;
+        s32 totalEntries;
+        f32 invW;
         s32 i;
+        s32 index;
+        f32 projectdPosScale;
+        f32 lockOnScaleX;
+        s32 triangleIndex;
 
         player = GET_PLAYER(play);
 
-        spCE = 0xFF;
-        var1 = 1.0f;
+        alpha = 255;
+        projectdPosScale = 1.0f;
 
-        if (targetCtx->unk_4B != 0) {
-            spB8 = 1;
+        if (targetCtx->rotZTick != 0) {
+            totalEntries = 1;
         } else {
-            spB8 = 3;
+            // Use multiple triangle sets for the movement effect when the triangles are
+            // getting closer to the actor from the margin of the screen
```

Comment:
maybe "Use multiple reticles for the effect of the reticle quickly zooming in on an actor from off screen"